sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/utils/AvroUtils.java

```diff
@@ -1277,4 +1308,17 @@ private static void checkTypeName(Schema.TypeName got, Schema.TypeName expected,
     checkArgument(
         got.equals(expected), "Can't convert '%s' to %s, expected: %s", label, got, expected);
   }
+
+  /** Helper factory to build JDBC Logical types for AVRO Schema. */
+  private static org.apache.avro.Schema makeJdbcLogicalStringAvroType(
+      Schema.LogicalType<?, ?> logicalType) {
+    JDBCType jdbcType = JDBCType.valueOf(logicalType.getIdentifier());
+    Integer size = logicalType.getArgument();
+
+    String schemaJson =
```

Comment:
Hey, optionally you can create Avro schemas in a more readable way:
```
return org.apache.avro.SchemaBuilder.builder()
    .stringBuilder().prop("logicalType", jdbcType.name()).prop("maxLength", size).endString();
```
This is for info -- they should (must!) be equivalent to using the Parser.  This snippet is small but builders are a good practice for larger schemas!

Reply:
@RyanSkraba thanks for the suggestion, one specific reason I didn't use the `SchemaBuilder` is due to the need to use a an Integer (non-String) value for the property. The Schema produced by using `SchemaBuilder` looks like:
`{"type":"string","logicalType":"LONGVARCHAR","maxLength": "50"}`
vs. the expected:
`{"type":"string","logicalType":"LONGVARCHAR","maxLength": 50}`

I think its the same reason, Hive's [TypeInfoToSchema#L116](https://github.com/apache/hive/blob/5d268834a5f5278ea76399f8af0d0ab043ae0b45/serde/src/java/org/apache/hadoop/hive/serde2/avro/TypeInfoToSchema.java#L116) also uses the JSON based parsing approach.

Reply:
Huh, I'm surprised -- thanks for pointing this out.  It might be an Avro bug!  I'm pretty sure that if size is an `int` it's a JSON number.
Regardless, thanks for the update, let's use the Hive approach then!

Reply:
I hate to be this annoying @anantdamle but since we are trying to 'align' with Hive/Spark maybe it is good that we name the logicalType names to coincide with the ones in the class you mention which are in lowercase.

https://github.com/apache/hive/blob/135629b8d6b538fed092641537034a9fbc59c7a0/serde/src/java/org/apache/hadoop/hive/serde2/avro/AvroSerDe.java#L59

Reply:
No worries @iemejia, concern in using is that Hive only provides `varchar`, how to then deal with others like longvarchar etc.
If I use lowercase then converting back to JDBCType would become hard.
Do you suggest converting all the string based logical types to just varchar with appropriate maxLength?

Reply:
All the logicalTypes that Hive/Spark define are spelled with lowercase so let's do like them.

It is a bit odd compared with the Java SQL Types (on uppercase), at least SQL should be casing agnostic so it should not matter in that front.

Reply:
@iemejia made the changes, with following mapping as per Hive, as there are only two categories:

**Variable Length Strings:**
[`VARCHAR`, `LONGVARCHAR`, `NVARCHAR`, `LONGNVARCHAR`] -> `varchar`

**Fixed Length Strings:**
[`CHAR`, `NCHAR`] -> `char`